rust/lance/src/dataset/cleanup.rs

```diff
@@ -394,61 +393,20 @@ pub async fn cleanup_old_versions(
     cleanup.run().await
 }
 
-/// Force cleanup of specific partial writes.
-///
-/// These files can be cleaned up easily with [cleanup_old_versions()] after 7 days,
-/// but if you know specific partial writes have been made, you can call this
-/// function to clean them up immediately.
-///
-/// To find partial writes, you can use the
-/// [crate::dataset::progress::WriteFragmentProgress] trait to track which files
-/// have been started but never finished.
-pub async fn cleanup_partial_writes(
-    store: &ObjectStore,
-    base_path: &Path,
-    objects: impl IntoIterator<Item = (&Path, &String)>,
-) -> Result<()> {
```

Comment:
Why get rid of this?  Was it no longer working?  Or no longer needed?

Reply:
`object_store` no longer exposes the multipart ids. So now users can't collect them and pass them into this function.

Instead, we automatically cleanup on `Drop` for `ObjectWriter`. In case of crashes, users should rely on setting a lifecycle rule to delete old incomplete writes. This is already recommended in our docs.